src/Network/Haskoin/Store/Data.hs

```diff
@@ -332,6 +368,13 @@ instance BinSerial Unspent where
         putWord64be v
         put $ B.Short.fromShort s
 
+    binDeserial net = do
+      b <- binDeserial net
+      p <- get
+      v <- getWord64be
+      s <- B.Short.toShort <$> get
```

Comment:
Conversely, here, `get` is enough, and if you wish you can implement it with applicative:

```haskell
Unspent
    <$> binDeserial net
    <*> get
    <*> getWord64be
    <*> get
```